File: employee-frontend/src/services/employeeService.js
import axios from "axios";

const BASE_URL = "http://localhost:8086/api/employees";

const employeeService = {
  // Get All Employees
  getAllEmployees: () => {
    return axios.get(BASE_URL);
  },

  // Get Employee By ID
  getEmployeeById: (id) => {
    return axios.get(`${BASE_URL}/${id}`);
  },

  // Create Employee
  createEmployee: (employee) => {
    return axios.post(BASE_URL, employee);
  },

  // Update Employee
  updateEmployee: (id, employee) => {
    return axios.put(`${BASE_URL}/${id}`, employee);
  },

  // Delete Employee
  deleteEmployee: (id) => {
    return axios.delete(`${BASE_URL}/${id}`);
  },
};

export default employeeService;

File: employee-backend/employee-backend/src/main/java/com/employee/employee_backend/service/EmployeeService.java
package com.employee.employee_backend.service;

import com.employee.employee_backend.model.Employee;
import com.employee.employee_backend.repository.EmployeeRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.Optional;

@Service
public class EmployeeService {

    @Autowired
    private EmployeeRepository employeeRepository;

    // ✅ Get All Employees
    public List<Employee> getAllEmployees() {
        return employeeRepository.findAll();
    }

    // ✅ Get Employee By ID
    public Optional<Employee> getEmployeeById(Long id) {
        return employeeRepository.findById(id);
    }

    // ✅ Create Employee
    public Employee createEmployee(Employee employee) {
        return employeeRepository.save(employee);
    }

    // ✅ Update Employee
    public Employee updateEmployee(Long id, Employee employeeDetails) {
        Employee employee = employeeRepository.findById(id)
                .orElseThrow(() -> new RuntimeException("Employee not found with id: " + id));

        employee.setFirstName(employeeDetails.getFirstName());
        employee.setLastName(employeeDetails.getLastName());
        employee.setEmail(employeeDetails.getEmail());
        employee.setDepartment(employeeDetails.getDepartment());
        employee.setPosition(employeeDetails.getPosition());
        employee.setSalary(employeeDetails.getSalary());

        return employeeRepository.save(employee);
    }

    // ✅ Delete Employee
    public void deleteEmployee(Long id) {
        Employee employee = employeeRepository.findById(id)
                .orElseThrow(() -> new RuntimeException("Employee not found with id: " + id));
        employeeRepository.delete(employee);
    }
}